pkg/util/buffered_chan_test.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package util
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestBufferedChan(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	c := NewBufferedChan(ctx, 10, 3)
+	go func() {
+		for n := 0; n < 100; n++ {
```

Comment:
To ensure that the assumptions are correct about the thread safety, I think this test should have a bigger `n` (1000/10000 maybe?) and have a semaphore on the put rather than starting immediately,

Reply:
I like the idea to have a bigger number.  The purpose of this test is to ensure that all values put by `Put` are retrieved by `Get` in the right order.   
It is unclear how a semaphore helps because there are 2 cases, when `Put` is called first or when `Get` is called first. Anyway, to ensure that the assumptions are correct about the thread safety, I rely  on `go test --race`.
Let me know if you think this is not enough.

Reply:
> It is unclear how a semaphore helps because there are 2 cases, when Put is called first or when Get is called first

In this case here, there's not a lot of assurance that `c.Put` ever occurs before `c.Get` because the subroutine can/may run and finish before c.Get loop gets called. Having a higher `n` and a semaphore will ensure that we get enough samples that we get both sets of conditions.

> I rely on go test --race

I think that reliance on `--race` should not be our default if we can create a test that reasonably covers the functionality without that flag (it's slow and I'm not really sure that it's deterministic).

Reply:
> In this case here, there's not a lot of assurance that c.Put ever occurs before c.Get

I did not make the assumption that `c.Put` is called before `c.Get` .  `c.Get`  blocks when there is no value to read.

Reply:
> I think that reliance on --race should not be our default if we can create a test that reasonably covers the functionality without that flag (it's slow and I'm not really sure that it's deterministic).

What tests do you have in mind? Do you think about having 2 tests, one where `c.Get` is called first and one where `c.Put` is called first? Do you think about having a test with random sleep before `c.Put`  and `c.Get` to simulate threads behavior?

Reply:
Unit test added.